github_workflow.md

```diff
@@ -0,0 +1,153 @@
+# GitHub Milestone and Issue Workflow
+
+This purpose of this document is to describe our development workflow on
+GitHub; in particular, how we use milestones and issues. We recognize that our
+many subprojects may use variations of these ideas. This particular workflow is
+based on the practices from the JupyterLab repo
+(https://github.com/jupyterlab/jupyterlab) and are optional for Jupyter
+subprojects to follow.
+
+## Milestones
+
+We use Github milestones to track issues and pull requests related to particular
+releases of a subproject (https://github.com/jupyterlab/jupyterlab/milestones).
+
+* These milestones should be labeled with the version numbers of the releases
+  (`4.3`, `5.0`, etc.).
+* Subprojects should only have a few numbered milestones. Beyond one or two
+  future releases, we have found that detailed milestones become meaningless.
+* Subprojects should also have a `Backlog` milestone for issues and pull
+  requests that have not been assigned to a particular release milestone. This
+  also includes new feature requests ("wishlist") issues that we haven't
+  committed to implementing. Essentially, `Backlog` means "at some point in the
+  future, maybe."
```

Comment:
Ah, that's an important distinction, and a pretty big change. Wishlist was used to manage attention - if it's in wishlist, it never needs immediate attention, but it can still a useful source of tasks. With backlog, everything with a numbered milestone is considered a 'blocker' for the release?

As we have worked in the past, a milestoned Issue is something that we need to at least consider, and explicitly decide *not* to address before making a release. I've found that to be a useful exercise, to make sure I'm always reminded of the things we are deciding not to do. Deciding certain issues aren't going to make this release is typically the first part of getting ready for a release. The backlog model does seem to better indicate how close we are to the possibility of a release at any point in time, but it also means the 'backlog' requires much more constant attention than wishlist did. That might be a good thing, I'm not sure.

I think there is some information missing about which milestone to apply to a confirmed issue and when. So far, I don't see anything that would make a difference in how I would assign Issues to milestones, and would end up treating backlog the same as wishlist.